lib/ThreeScaleClient.php

```diff
@@ -95,7 +110,18 @@ public function getHost() {
    */
   public function authorize($appId, $appKey = null, $serviceId = null, $usage = null) {
     $url = "http://" . $this->getHost() . "/transactions/authorize.xml";
-    $params = array('provider_key' => $this->getProviderKey(), 'app_id' => $appId);
+    $params = array('app_id' => $appId);
+
+    $providerKey = $this->getProviderKey();
+    $serviceToken = $this->getServiceToken();
```

Comment:
If you can configure service token just globally, you can use this client only with ONE service.

I'd consider passing the service token on per call basis to override the global one. But it depends if other plugins follow this (Ruby one does).